app/models/API1812/latePaymentPenalty/LatePaymentPenalty.scala

```diff
@@ -35,30 +34,97 @@ case class LatePaymentPenalty(principalChargeReference: String,
                               penaltyChargeReference: Option[String],
                               timeToPay: Option[Seq[TimeToPay]]) {
 
-  def hasTimeToPay(implicit appConfig: AppConfig): Boolean = {
-    timeToPay.fold(false)(_.exists(ttp => ttp.TTPStartDate.exists(!_.isAfter(DateService.now))
-      && ttp.TTPEndDate.exists(!_.isBefore(DateService.now))))
-  }
+  def hasTimeToPay(implicit appConfig: AppConfig): Boolean =
+    timeToPay.fold(false)(
+      _.exists(ttp =>
+        ttp.TTPStartDate.exists(!_.isAfter(DateService.now))
+          && ttp.TTPEndDate.exists(!_.isBefore(DateService.now))))
 }
 
 object LatePaymentPenalty {
 
   implicit val reads: Reads[LatePaymentPenalty] = Json.reads[LatePaymentPenalty]
 
   implicit def writes(implicit appConfig: AppConfig): Writes[LatePaymentPenalty] = { model =>
-    JsObject(Json.obj(
-      "principalChargeReference" -> model.principalChargeReference,
-      "penaltyCategory" -> model.penaltyCategory,
-      "LPP1LRCalculationAmount" -> model.LPP1LRCalculationAmount,
-      "LPP1LRDays" -> model.LPP1LRDays,
-      "LPP1LRPercentage" -> model.LPP1LRPercentage,
-      "LPP1HRCalculationAmount" -> model.LPP1HRCalculationAmount,
-      "LPP1HRDays" -> model.LPP1HRDays,
-      "LPP1HRPercentage" -> model.LPP1HRPercentage,
-      "LPP2Days" -> model.LPP2Days,
-      "LPP2Percentage" -> model.LPP2Percentage,
-      "penaltyChargeReference" -> model.penaltyChargeReference,
-      "timeToPay" -> model.hasTimeToPay
-    ).fields.filterNot(_._2 == JsNull))
+    JsObject(
+      Json
+        .obj(
+          "principalChargeReference" -> model.principalChargeReference,
+          "penaltyCategory"          -> model.penaltyCategory,
+          "LPP1LRCalculationAmount"  -> model.LPP1LRCalculationAmount,
+          "LPP1LRDays"               -> model.LPP1LRDays,
+          "LPP1LRPercentage"         -> model.LPP1LRPercentage,
+          "LPP1HRCalculationAmount"  -> model.LPP1HRCalculationAmount,
+          "LPP1HRDays"               -> model.LPP1HRDays,
+          "LPP1HRPercentage"         -> model.LPP1HRPercentage,
+          "LPP2Days"                 -> model.LPP2Days,
+          "LPP2Percentage"           -> model.LPP2Percentage,
+          "penaltyChargeReference"   -> model.penaltyChargeReference,
+          "timeToPay"                -> model.hasTimeToPay
+        )
+        .fields
+        .filterNot(_._2 == JsNull))
   }
 }
+
```

Comment:
To match the [schema](https://admin.tax.service.gov.uk/integration-hub/apis/view-specification/79209ad9-54ea-4044-9115-b1b077bc223c/test#tag/List-of-Penalties/operation/getPenalties)
<img width="681" height="699" alt="image" src="https://github.com/user-attachments/assets/90665223-4c9b-4f28-a502-004e887fb1a0" />

This then get's mapped to the old version which is still used by our frontends and by ITSA.
Because it is shared with ITSA, it is unlikely we will change this during the clean up, we will more likely continue using the mapping to the old model.